src/main/java/frc/robot/Drivetrain.java

```diff
@@ -47,63 +47,63 @@
   private static final int kEncoderResolution = 2048; // falcon 500 + talon FX encoder
   private static final double kEncoderTicksPerMeter = (50.0/12.0) * kEncoderResolution / ( 2.0*kWheelRadius*Math.PI ); // replace 12 with real Motor:Wheel gear Ratio
 
   private final WPI_TalonFX m_leftLeader = new WPI_TalonFX(11);
   private final WPI_TalonFX m_rightLeader = new WPI_TalonFX(12);
 
-  // private final SpeedControllerGroup m_leftGroup =
-  // new SpeedControllerGroup(m_leftLeader, m_leftFollower);
-  // private final SpeedControllerGroup m_rightGroup =
-  // new SpeedControllerGroup(m_rightLeader, m_rightFollower);
+  private final PIDController m_leftPIDController = new PIDController(1, 0, 0);
+  private final PIDController m_rightPIDController = new PIDController(1, 0, 0);
 
-  // private final Encoder m_leftEncoder = new Encoder(0, 1);
-  // private final Encoder m_rightEncoder = new Encoder(2, 3);
-  // replaced with motor.getposition calls
-
-  private final PIDController m_leftPIDController = new PIDController(2.5, 0, 0);
-  private final PIDController m_rightPIDController = new PIDController(2.5, 0, 0);
-
-  // private final AnalogGyro m_gyro = new AnalogGyro(0);
-  private final AHRS m_gyro = new AHRS(SPI.Port.kMXP);
+  private AHRS m_gyro = new AHRS(SPI.Port.kMXP);
+  private final AnalogGyro m_analog_gyro = new AnalogGyro(0);
 
   private final DifferentialDriveKinematics m_kinematics = new DifferentialDriveKinematics(kTrackWidth);
-  private final DifferentialDriveOdometry m_odometry = new DifferentialDriveOdometry(m_gyro.getRotation2d());
+  private DifferentialDriveOdometry m_odometry;
 
-  // Gains are for example purposes only - must be determined for your own
-  // robot!
-  private final SimpleMotorFeedforward m_feedforward = new SimpleMotorFeedforward(0.1, 0.25);
+  // Gains are for example purposes only - must be determined for your own robot!
+  private final SimpleMotorFeedforward m_feedforward = new SimpleMotorFeedforward(1, 2, 1);
 
   // Simulation classes help us simulate our robot
-  // private final AnalogGyroSim m_gyroSim = new AnalogGyroSim(m_gyro);
-  // private final EncoderSim m_leftEncoderSim = new EncoderSim(m_leftEncoder);
-  // private final EncoderSim m_rightEncoderSim = new EncoderSim(m_rightEncoder);
-  // private final Field2d m_fieldSim = new Field2d();
-  // private final LinearSystem<N2, N2, N2> m_drivetrainSystem = LinearSystemId.identifyDrivetrainSystem(1.98, 0.2, 1.5,
-  //     0.3);
-  // private final DifferentialDrivetrainSim m_drivetrainSimulator = new DifferentialDrivetrainSim(m_drivetrainSystem, 
-  //     DCMotor.getFalcon500(1), 12, kTrackWidth, kWheelRadius, null);
+  private double leftVolt,rightVolt;
+  private final Encoder m_leftEncoder = new Encoder(0, 1);
+  private final Encoder m_rightEncoder = new Encoder(2, 3);
+  private final AnalogGyroSim m_gyroSim = new AnalogGyroSim(m_analog_gyro);
+  private final EncoderSim m_leftEncoderSim = new EncoderSim(m_leftEncoder);
```

Comment:
move the actual declarations of the Sim-Only members that reference sensors not on the real robot into the Simulation check in the constructor.
```suggestion
  private double leftVolt,rightVolt;
  private final Encoder m_leftEncoder;
  private final Encoder m_rightEncoder;
  private final AnalogGyroSim m_gyroSim;
  private final EncoderSim m_leftEncoderSim;
```